libraries/ESP8266WiFi/src/ESP8266WiFiSTA.cpp

```diff
@@ -121,7 +121,9 @@ wl_status_t ESP8266WiFiSTAClass::begin(const char* ssid, const char *passphrase,
     conf.threshold.authmode = AUTH_OPEN;
 
     if(passphrase) {
-        conf.threshold.authmode = AUTH_WPA_PSK;
+        if (!_useInsecureWEP) {

```

Comment:
This is missing the else, and un-reverse the logic: Something like this:
`conf.threshold.authmode = _useInsecureWEP ?  AUTH_WEP : AUTH_WPA`;

The reason why the else is needed is to avoid having the ESP bounce off the intended AP and connect to an open AP. In other words, if a pass is provided, avoid connecting to any open AP.

Reply:
done